app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,13 +11,21 @@ def self.call(*args)
       end
 
       def call
-        # This method should find the user based on the `receiver_id`.
-        # It should then determine the appropriate Broadcast for a user,
-        # based on the `created_at` and the different conditions for sending a notification.
-        # `welcome_broadcast = ...`
+        return if commented_on_welcome_thread? || received_notification?
 
-        # Once it has the appropriate Broadcast to be sent, it should send a notification for it:
-        # `Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)`
+        welcome_broadcast = Broadcast.find_by(title: "Welcome Notification: welcome_thread")
+        Notification.send_welcome_notification(receiver_id, welcome_broadcast.id)
+      end
+
+      def received_notification?
+        welcome_broadcast = Broadcast.find_by(title: "Welcome Notification: welcome_thread")
+        Notification.exists?(notifiable_id: welcome_broadcast.id, user_id: receiver_id)
```

Comment:
I think you could also do `notifiable: welcome_broadcast` here, instead of `notifiable_id: welcome_broadcast.id`!